libnavigation-core/src/main/java/com/mapbox/navigation/core/arrival/ArrivalController.kt

```diff
@@ -40,7 +42,13 @@ class AutoArrivalController : ArrivalController {
      * By default this will move onto the next step.
      */
     override fun navigateNextRouteLeg(routeLegProgress: RouteLegProgress): Boolean {
-        return true
+        return if (count > 0) {
+            count--
+            false
+        } else {
+            count = arrivalOptions().arrivalInSeconds?.toInt() ?: 0
```

Comment:
Can we rely on `ArrivalOptions` in all situations? What happens if a new route is set? Are we resetting the `count`? Won't the current `count` be carried from the previous one? 🤔

Reply:
hmm the experience would be inconsistent when changing routes during arrival. But you wouldn't notice it until your next arrival

Reply:
if RouteLegProgress had a unique identifier, this could be a simple solution.